core/trino-spi/src/test/java/io/trino/spi/type/TestDecimalAggregationAccumulatorType.java

```diff
@@ -0,0 +1,154 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.type;
+
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+import io.trino.spi.block.Block;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.block.DecimalAggregationAccumulatorBlock;
+import io.trino.spi.block.DecimalAggregationAccumulatorBlockBuilder;
+import org.testng.annotations.Test;
+
+import static io.trino.spi.type.DecimalAggregationAccumulatorType.LONG_DECIMAL_WITH_OVERFLOW_AND_LONG;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+
+public class TestDecimalAggregationAccumulatorType
+{
+    private static final Type TYPE = LONG_DECIMAL_WITH_OVERFLOW_AND_LONG;
+    private static final int OVERFLOW_OFFSET = 3 * Long.BYTES;
+
+    @Test
+    public void testWriteRead()
+    {
+        BlockBuilder blockBuilder = TYPE.createBlockBuilder(null, 2);
+        Slice entry = Slices.wrappedLongArray(1, 2, 3, 4);
+        TYPE.writeSlice(blockBuilder, entry);
+        blockBuilder.appendNull();
+        Block block = blockBuilder.build();
+
+        testWriteRead(entry, block);
+        testWriteRead(entry, blockBuilder);
+    }
+
+    private void testWriteRead(Slice entry, Block block)
+    {
+        assertEquals(block.getPositionCount(), 2);
+        assertEquals(block.getLong(0, 0), 1);
+        assertEquals(block.getLong(0, Long.BYTES), 2);
+        assertEquals(block.getLong(0, Long.BYTES * 2), 3);
+        assertEquals(block.getLong(0, Long.BYTES * 3), 4);
+        assertEquals(TYPE.getSlice(block, 0), entry);
+        assertEquals(TYPE.getObjectValue(null, block, 0), new SqlVarbinary(entry.getBytes()));
+
+        assertTrue(block.isNull(1));
+        assertNull(TYPE.getObjectValue(null, block, 1));
+
+        BlockBuilder newBlockBuilder = TYPE.createBlockBuilder(null, 2);
+        TYPE.appendTo(block, 0, newBlockBuilder);
+        TYPE.appendTo(block, 1, newBlockBuilder);
+        Block newBlock = newBlockBuilder.build();
+        assertEquals(TYPE.getSlice(block, 0), TYPE.getSlice(newBlock, 0));
+        assertNull(TYPE.getObjectValue(null, newBlock, 1));
+        assertTrue(newBlock.isNull(1));
+    }
+
+    @Test
+    public void testRegion()
+    {
+        int entries = 4;
+        BlockBuilder blockBuilder = TYPE.createBlockBuilder(null, entries);
+        TYPE.writeSlice(blockBuilder, entry(1));
+        blockBuilder.appendNull();
+        TYPE.writeSlice(blockBuilder, entry(2));
+        TYPE.writeSlice(blockBuilder, entry(3));
+        Block block = blockBuilder.build();
+
+        testRegion(block);
+        testRegion(blockBuilder);
+    }
+
+    private void testRegion(Block block)
```

Comment:
how this test is specific to type? It seems like this should be `DecimalAggregationAccumulatorBlock/DecimalAggregationAccumulatorBlockBuilder` test, e.g: `TestDecimalAggregationAccumulatorBlock`

`LONG_DECIMAL_WITH_OVERFLOW_AND_LONG` seems like it's a test helper here